src/community/params-extractor/src/main/java/org/geoserver/params/extractor/Filter.java

```diff
@@ -0,0 +1,63 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.params.extractor;
+
+import org.geoserver.filters.GeoServerFilter;
+import org.geoserver.platform.ExtensionPriority;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.ResourceStore;
+import org.geotools.util.logging.Logging;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+public final class Filter implements GeoServerFilter, ExtensionPriority {
+
+    private static final Logger LOGGER = Logging.getLogger(Filter.class);
+
+    private static List<Rule> rules;
+
+    public Filter(ResourceStore dataDirectory) {
+        Resource resource = dataDirectory.get("rules.xml");
```

Comment:
Is the rules.xml stored in main data_dir folder? Can we move it to a more meaningful subfolder? (params-extractor) or give it a more meaningful name than rules.xml (params-extractor.xml)?